### PR TITLE
325 walletconnect v2

### PR DESCRIPTION
**Pull request by L03TJ3**
# Description
Web3onboard has released the official package with wc-v2 support.
This PR implements it, also to test out with good-wallet compatibility

- [x] - update customwalletconnect to support or enable sdk-v2

About # (link your issue here)
https://github.com/GoodDollar/GoodProtocolUI/issues/325

# How Has This Been Tested?
WIP on local

Please describe the tests that you ran to verify your changes.

# Checklist:
- [ ] PR title matches follow: (Feature|Bug|Chore) Task Name
- [ ] My code follows the style guidelines of this project
- [ ] I have followed all the instructions described in the initial task (check Definitions of Done)
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added reference to a related issue in the repository
- [ ] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: packages/sdk-v2/src/sdk/faucet/react.ts [hunk 1]
```diff
@@ -55,7 +55,7 @@ export const useFaucet = async (refresh: QueryParams["refresh"] = 12) => {
     const [canTop, toppingAmount] = faucetResult.map(_ => _?.value?.[0] as boolean | BigNumber | undefined) || [];
     const threshold = (toppingAmount as BigNumber)?.mul(50)?.div(100) || minBalance;
 
-    console.log("faucet:", { canTop, balance, threshold, account, toppingAmount });
+    // console.log("faucet:", { canTop, balance, threshold, account, toppingAmount });
```
Comment (johnsmith-gooddollar) [inline packages/sdk-v2/src/sdk/faucet/react.ts:58]:
```suggestion
```

File: packages/sdk-v2/src/sdk/claim/react.ts [hunk 2]
```diff
@@ -175,6 +175,7 @@ export const useWhitelistSync = () => {
     };
 
     whitelistSync().catch(noop);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (johnsmith-gooddollar) [inline packages/sdk-v2/src/sdk/claim/react.ts:178]: why not add proper deps ?
if you have issue with both [value, setValue] needed causing infinite effect indication, you could omit it by passing update callback to setter e.g.

```javascript
setValue(value => <calculate new value from value>)
```

OR/AND using refs for some flag-like props (e.g. syncInProgress). Just do not forgot to update those refs values via separate effect

**Review by johnsmith-gooddollar — CHANGES_REQUESTED** [review]

File: packages/sdk-v2/src/sdk/faucet/react.ts (hunk 1 above)
Comment (L03TJ3) [reply packages/sdk-v2/src/sdk/faucet/react.ts]: removed

File: packages/sdk-v2/src/sdk/claim/react.ts (hunk 2 above)
Comment (L03TJ3) [reply packages/sdk-v2/src/sdk/claim/react.ts]: The rule has been removed, so these comments are not necessary anymore.
But after some testing and moving baseEnv to a ref I enabled the rule and the warning disappeared altogether

Also fixed a logic-error when its trying to find the backend url

**Review by johnsmith-gooddollar — APPROVED** [review]

**Pull request merged**